code/game/gamemodes/cult/runes.dm

```diff
@@ -831,110 +680,54 @@ var/list/teleport_other_runes = list()
 		log_game("Talisman Imbue rune failed - no blank papers on rune")
 		return
 	var/obj/item/weapon/paper/paper_to_imbue = pick(papers_on_rune)
-	var/list/nearby_runes = list()
-	for(var/obj/effect/rune/R in orange(1,src))
-		nearby_runes.Add(R)
-	if(!nearby_runes.len)
-		user << "<span class='cultitalic'>There are no runes near [src]!</span>"
-		fail_invoke()
-		log_game("Talisman Imbue rune failed - no nearby runes")
-		return
-	var/obj/effect/rune/picked_rune = pick(nearby_runes)
-	var/list/split_rune_type = splittext("[picked_rune.type]", "/")
-	var/imbue_type = split_rune_type[split_rune_type.len]
-	var/talisman_type = text2path("/obj/item/weapon/paper/talisman/[imbue_type]")
-	if(ispath(talisman_type))
-		user.say("H'drak v'loso, mir'kanas verbot!")
-		var/obj/item/weapon/paper/talisman/TA = new talisman_type(get_turf(src))
-		if(istype(picked_rune, /obj/effect/rune/teleport))
-			var/obj/effect/rune/teleport/TR = picked_rune
-			var/obj/item/weapon/paper/talisman/teleport/TT = TA
-			TT.keyword = TR.keyword
-	else
-		user << "<span class='cultitalic'>The chosen rune was not a valid target!</span>"
-		fail_invoke()
-		log_game("Talisman Imbue rune failed - chosen rune invalid")
+	for(var/I in subtypesof(/obj/item/weapon/paper/talisman) - /obj/item/weapon/paper/talisman/malformed - /obj/item/weapon/paper/talisman/supply)
+		var/obj/item/weapon/paper/talisman/J = I
+		if(initial(J.cultist_name))
+			possible_talismans.Add(initial(J.cultist_name)) //This is to allow the menu to let cultists select talismans by name
+	entered_talisman_name = input(user, "Choose a talisman to imbue.", "Talisman Choices") as null|anything in possible_talismans
+	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated())
 		return
-	visible_message("<span class='warning'>[picked_rune] crumbles to dust, and bloody images form themselves on [paper_to_imbue].</span>")
+	for(var/I in typesof(/obj/item/weapon/paper/talisman))
+		var/obj/effect/rune/J = I
+		if(initial(J.cultist_name) == entered_talisman_name)
+			talisman_type = J
+	user.say("H'drak v'loso, mir'kanas verbot!")
+	visible_message("<span class='warning'>Dark power begins to channel into the paper!.</span>")
+	if(!do_after(user, 100, target = get_turf(user)))
+		return
+	var/obj/item/weapon/paper/talisman/TA = new talisman_type(get_turf(src))
+	if(istype(TA, /obj/item/weapon/paper/talisman/teleport))
+		var/the_keyword = stripped_input(usr, "Please enter a keyword for the talisman.", "Enter Keyword", "")
+		var/obj/item/weapon/paper/talisman/teleport/TELE = TA
+		TELE.keyword = the_keyword
+	visible_message("<span class='warning'>[src] crumbles to dust, and bloody images form themselves on [paper_to_imbue].</span>")
 	qdel(paper_to_imbue)
-	qdel(picked_rune)
 	qdel(src)
```

Comment:
This is probably bad just from how much of a delay this causes; if you want multiple talismans this is like 15 seconds for each, 10 from talisman creation and 5 from rune scribing, less if you move off the rune first.

This makes stuff like armor and summon tome absolute butts to do because it's a huge delay, you still need paper, and you don't even have the option to make the tome/armor rune itself.